retnet/modeling_retnet.py

```diff
@@ -841,13 +1046,15 @@ def forward(
             hidden_states = F.pad(hidden_states, (0, 0, 0, padding_len))
         else:
             slen = seq_length
+        if fixed_seq_len:slen=fixed_seq_len
         # relative position
         if retention_rel_pos is None:
             retention_rel_pos = self.retnet_rel_pos(slen,
                                                     forward_impl=forward_impl,
                                                     recurrent_chunk_size=recurrent_chunk_size,
                                                     retention_mask=retention_mask,
-                                                    get_decay_scale=not self.training)
```

Comment:
Why do we want decay scale during training?

Reply:
Below is an example for `one parallel output` and `one recurrent output`
```
with torch.inference_mode(): #<--almost equal to `torch.no_grad()`
   model.eval() # <-- this disable dropout and batchnorm or other layers that behave differently during inference
   out = model(old_inputs,
      forward_impl='parallel' #<--  this line indicates parallel mode 
      use_cached=True,#<--  must have use_cached = True  
      **args,)
   past_kv = out.past_key_values
   model.train() # if want train later token, one need reactivate it here.
   out = model(new_inputs,
      forward_impl='recurrent' #<--  this line indicates recurrent mode 
      use_cached=True,#<--  must have use_cached = True for further recurrent mode
      past_key_values=past_kv, # this line must
      **args
   )
```
If we don't have the `model.eval()` the recurrent mode fail to generate by take the `scale=None`.
However, the `model.eval()` will change the behavior of some layer. 

There is no other important reason here, just for me convenience.
----
Basically, the goal is to generate a `cache` first and reuse it many times
cache --> task_1
cache --> task_2
---